src/Symfony/Component/Cache/CacheItem.php

```diff
@@ -100,6 +100,30 @@ public function expiresAfter($time)
     }
 
     /**
+     * Validates a cache key according to PSR-6.
+     *
+     * @param string $key The key to validate.
+     *
+     * @return string $key if it is valid.
+     *
+     * @throws InvalidArgumentException When $key is not valid.
+     */
+    public static function validateKey($key)
```

Comment:
@javiereguiluz but your suggestion of a true/false validator is precisely what makes things much harder to use: when receiving false, you cannot know what is the reason, and so you cannot put a meaningful message in your exception. And if you keep throwing inside the validator, it is not a true/false one anymore but a true/Exception one (with a name ``isValid`` which still make people think it is a true/false one). In this case, I would argue that a void/Exception signature is better than a true/Exception one.